internal/httperror/error.go

```diff
@@ -47,3 +48,8 @@ func Is(err error, code int) bool {
 func IsNotFound(err error) bool {
 	return Is(err, http.StatusNotFound)
 }
+
+func IsNetworkError(err error) bool {
+	netErr, ok := err.(net.Error)
+	return ok && netErr.Timeout()
+}
```

Comment:
Two things here:
1. I think `IsTimeoutError` would be a more fitting name
2. Plase use [`errors.As`](https://pkg.go.dev/errors#As) instead of a simple cast. It is more resilient and also checks wrapped errors. Something like this should work:
```go
var netErr net.Error 
return errors.As(err, &netErr) && netErr.Timeout()
```